### PR TITLE
Fix bug on wallet list modal where if wallet is created then tries to select it after but created wallet is not yet on the redux state.

### PR DESCRIPTION
**Pull request by amanalili**
#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS Tablet
- [ ] Tested on small Android

### CODE REVIEW

File: src/components/modals/WalletListModal.js [hunk 1]
```diff
@@ -292,7 +292,15 @@ class WalletListModalConnected extends React.Component<Props, State> {
   createAndSelectWallet = async ({ currencyCode, walletType }: CreateWalletType) => {
     try {
       const wallet = await this.createWallet(currencyCode, walletType)
-      this.props.bridge.resolve({ walletId: wallet.id, currencyCode: wallet.currencyInfo.currencyCode })
+      const checkWalletsThenSelect = async () => {
```
Comment (paullinator) [inline src/components/modals/WalletListModal.js:295]: No need to make this an async function.

File: src/components/modals/WalletListModal.js [hunk 2]
```diff
@@ -292,7 +292,15 @@ class WalletListModalConnected extends React.Component<Props, State> {
...
+        if (this.props.wallets[wallet.id]) {
+          this.props.bridge.resolve({ walletId: wallet.id, currencyCode: wallet.currencyInfo.currencyCode })
+        } else {
+          await setTimeout(() => {}, 50)
```
Comment (paullinator) [inline src/components/modals/WalletListModal.js:299]: No need to await. Just set `checkWalletsThenSelect` as the callback of setTimeout.

**Review by paullinator — CHANGES_REQUESTED** [review]

**Comment by amanalili** [comment]
Fixed

File: src/components/modals/WalletListModal.js [hunk 3]
```diff
@@ -292,7 +292,14 @@ class WalletListModalConnected extends React.Component<Props, State> {
   createAndSelectWallet = async ({ currencyCode, walletType }: CreateWalletType) => {
     try {
       const wallet = await this.createWallet(currencyCode, walletType)
-      this.props.bridge.resolve({ walletId: wallet.id, currencyCode: wallet.currencyInfo.currencyCode })
+      const checkWalletsThenSelect = () => {
+        if (this.props.wallets[wallet.id]) {
+          this.props.bridge.resolve({ walletId: wallet.id, currencyCode: wallet.currencyInfo.currencyCode })
+        } else {
+          setTimeout(() => checkWalletsThenSelect(), 50)
```
Comment (paullinator) [inline src/components/modals/WalletListModal.js:299]: Why do you have the anonymous function? just put checkWalletsThenSelect as the callback

**Review by paullinator — CHANGES_REQUESTED** [review]

**Review by paullinator — APPROVED** [review]

**Pull request merged**